### PR TITLE
Detect verify collisions and bail if so

### PR DESCRIPTION
**Pull request by stefwalter**
Abort testing if another verify machine is testing the same revision and has "won" the race to update Github.

### CODE REVIEW

**Comment by stefwalter** [comment]
Depends on #3178

**Comment by mvollmer** [comment]
#3178 is in master.

File: test/check-verify [hunk 1]
```diff
@@ -201,7 +205,17 @@ def main():
...
-        ret = step(opts, sink)
+        if opts.rebase:
+            rebase(sink)
+        if opts.install:
+            install(opts)
+
+        # Check if we are still publishing, in case some other verify
+        # machine collided with our test
+        if opts.github and sink:
+            check_publishing(sink, github)
```
Comment (mvollmer) [inline test/check-verify]: Why is check_publish called here, and not earlier?  So that the github status has time to settle down?

Comment (stefwalter) [inline test/check-verify]: Yes, we use the build delay as that settling period.

**Pull request closed without merging**